src/notebooks/record_dedupe.py

```diff
@@ -0,0 +1,111 @@
+from typing import Callable
+import pandas as pd
+
+PATH_TO_DATA = '/Users/Brandon/Downloads/Archive/Patient-1.ndjson'
+
+def make_test_dedupe_data(df: pd.DataFrame):
+    '''
+    Turn a bundle of patient data into a small test set for verifying
+    efficacy of deduplication. There will be 6 unique records here,
+    4 of which are completely dissimilar, one which will have three
+    duplicated versions of varying changes, and one which will be a 
+    near match for the duplicates but is off by one character.
+    '''
+    
+    # Full match case: everything the same, should be no merge conflicts
+    df.loc[6, :] = df.loc[0, :]
+
+    # Most match case: one field blank, one present
+    df.loc[7, :] = df.loc[0, :]
+    df.loc[7, 'gender'] = ''
+
+    # Multi-match case: store both in ndjson
+    df.loc[8, :] = df.loc[0, :]
+    df.loc[8, 'address'] = df.loc[10, 'address']
+
+    # Near-match case: not actually a dupe, but off by one char
+    df.loc[9, :] = df.loc[0, :]
+    dob_list = df.loc[9, 'birthDate'].split('-')
+    dob_list[1] = str((int(dob_list[1]) + 1) % 12)
+    df.loc[9, 'birthDate'] = '-'.join(dob_list)
+
+    return df[:10]
+
```

Comment:
In general I'm not a huge fan of selecting columns by their index because if there the order of columns in the data ever changes everything breaks. Using columns names can still break, but it slightly more robust. Not a big deal here given that this func is just making test data for us.

Reply:
Small note that none of the numbers in this function select columns. The first field of a `.loc[]` access corresponds to the index of a dataframe, which is almost always the row. Column names are specified here by name, which at least in my experience is normal convention (curious why you only regard it as "slightly" more robust? It seems to be the most widely used method of accessing columns, particularly because specifying particular column schemas when people enter data is also par for the course; unless you want to implement fuzzy matching on just column names, which seems like an absolute _nightmare_ 🤮 )

Reply:
ooops I read the indexing too quickly. It is definitely the most robust thing we can do, but it breaks when column names changes even though it dose not depend on column order which indexing columns dose. I have had problems in the past where there is some template csv that is suppose to be used, but then folks screw with things. This probably won't be an issue for us given the initial conversion to FHIR.